src/braket/circuits/circuit.py

```diff
@@ -877,6 +877,67 @@ def to_ir(self) -> Program:
             basis_rotation_instructions=ir_basis_rotation_instructions,
         )
 
+    @staticmethod
+    def _einsum_subscripts(targets: QubitSet, qubit_count: int) -> str:
+        target_count = len(targets)
+
+        gate_left_indexes = list(range(target_count))
+        un_left_indexes = list(range(target_count, target_count + qubit_count))
+        un_right_indexes = list(range(target_count + qubit_count, target_count + 2 * qubit_count))
+
+        gate_right_indexes = [un_left_indexes[-1 - target] for target in targets]
+
+        result_left_indexes = un_left_indexes.copy()
+        for pos, target in enumerate(targets):
+            result_left_indexes[-1 - target] = gate_left_indexes[pos]
+
+        return (
+            gate_left_indexes + gate_right_indexes,
+            un_left_indexes + un_right_indexes,
+            result_left_indexes + un_right_indexes,
+        )
+
+    def as_unitary(self) -> np.ndarray:
+        """
+        Unitary matrix representation of the entire circuit.
+        Notice that noise operations are not part of the returned matrix. Also,
+        any circuit with 10 qubits or more can get slow.
```

Comment:
Having this comment is great. What meant in the last PR thought was that we should print "Calling as_unitary() for circuits with more than 10 qubits can get slow." if  number of qubits is detected to be larger than 10.

Reply:
Done in https://github.com/aws/amazon-braket-sdk-python/pull/253/commits/b916246c779109796211b6d19251bc07b1018056